File: sloth-services/sloth.Application/Models/Jobs/GetBugItem.cs
using sloth.Domain.Entities;

namespace sloth.Application.Models.Jobs;
public class GetBugItem
{
    public int JobID { get; set; }
    public string Header { get; set; } = default!;
    public string Description { get; set; } = default!;
    public string Type { get; set; } = default!;
    public DateTime CreatedDate { get; set; }
    public DateTime? UpdatedDate { get; set; }
    public DateTime? ClosedDate { get; set; } = null;
    public bool IsClosed { get; set; } = false;
    public bool IsBlocker { get; set; } = false;
    public GetUserBugItem? CreatedBy { get; set; } = null;
    public GetUserBugItem? ClosedBy { get; set; } = null;
    public GetClientBugItem? Client { get; set; } = null;
    public GetUserBugItem? UpdatedBy { get; set; } = null;
    public GetPriorityBugItem? Priority { get; set; } = null;
    public GetStatusBugItem? Status { get; set; } = null;
    public IEnumerable<GetAssignmentBugItem> Assignments { get; set; } = [];
    public IEnumerable<GetChildJobLinkBugItem> ChildJobs { get; set; } = [];
    public IEnumerable<GetCommentBugItem> Comments { get; set; } = [];
    public IEnumerable<GetFileBugItem> Files { get; set; } = [];
    public IEnumerable<GetFunctionalityBugItem> Functionalities { get; set; } = [];
    public IEnumerable<JobHistory> History { get; set; } = [];
    public IEnumerable<GetParentJobLinkBugItem> ParentJobs { get; set; } = [];
    public IEnumerable<GetProductBugItem> Products { get; set; } = [];
}
public class GetAssignmentBugItem
{
    public DateTime AssignedDate { get; set; }
    public string AssignedTo { get; set; } = default!;
    public string AssignedToFullName { get; set; } = default!;
    public string AssignedToEmail { get; set; } = default!;
    public string AssignedBy { get; set; } = default!;
    public string AssignedByFullName { get; set; } = default!;
    public string AssignedByEmail { get; set; } = default!;
}
public class GetClientBugItem
{
    public Guid ClientID { get; set; } = default!;
    public string Name { get; set; } = default!;
    public string Alias { get; set; } = default!;
}
public class GetCommentBugItem
{
    public int CommentID { get; set; }
    public string Comment { get; set; } = default!;
    public DateTime CommentDate { get; set; } = default!;
    public bool IsEdited { get; set; } = false;
    public string CommentedBy { get; set; } = default!;
    public string CommentedByEmail { get; set; } = default!;
    public string CommentedByFullName { get; set; } = default!;
}
public class GetFileBugItem
{
    public Guid FileID { get; set; }
    public string Name { get; set; } = default!;
    public long Size { get; set; }
    public string Extension { get; set; } = default!;
    public DateTime AddedDate { get; set; }
    public string AddedBy { get; set; } = default!;
    public string AddedByEmail { get; set; } = default!;
    public string AddedByFullName { get; set; } = default!;
}
public class GetFunctionalityBugItem
{
    public int FunctionalityID { get; set; }
    public string Name { get; set; } = default!;
    public string Tag { get; set; } = default!;
    public string? TagColor { get; set; } = default;
    public string Description { get; set; } = default!;
}
public class GetPriorityBugItem
{
    public int PriorityID { get; set; }
    public string Tag { get; set; } = default!;
    public string? TagColor { get; set; } = default;
    public string? Description { get; set; } = null;
}
public class GetProductBugItem
{
    public int ProductID { get; set; }
    public string Alias { get; set; } = default!;
    public string Name { get; set; } = default!;
    public string? Description { get; set; } = null;
}
public class GetStatusBugItem
{
    public int StatusID { get; set; }
    public string Tag { get; set; } = default!;
    public string? TagColor { get; set; } = default;
    public string? Description { get; set; } = null;
}
public class GetTeamBugItem
{
    public string Alias { get; set; } = default!;
    public string Speciality { get; set; } = default!;
    public string Name { get; set; } = default!;
    public string Description { get; set; } = default!;
}
public class GetUserBugItem
{
    public string UserName { get; set; } = default!;
    public string FirstName { get; set; } = default!;
    public string LastName { get; set; } = default!;
    public string FullName { get; set; } = default!;
    public string Email { get; set; } = default!;
}
public class GetJobLinkBugItem
{
    public int JobID { get; set; } = default;
    public string JobHeader { get; set; } = default!;
    public string JobDescription { get; set; } = default!;
    public DateTime LinkDate { get; set; } = default!;
    public string LinkedBy { get; set; } = default!;
    public string LinkedByFullName { get; set; } = default!;
    public string LinkedByEmail { get; set; } = default!;
}
public class GetParentJobLinkBugItem : GetJobLinkBugItem;
public class GetChildJobLinkBugItem : GetJobLinkBugItem;

File: sloth-client/projects/sloth-http/src/lib/dto/job/items/get-bug.item.ts
export interface GetBugItem {
    jobID: number;
    header: string;
    description: string;
    type: string;
    createdDate: Date;
    updatedDate: Date | null;
    closedDate: Date | null;
    isClosed: boolean;
    isBlocker: boolean;
    createdBy: GetUserBugItem | null;
    closedBy: GetUserBugItem | null;
    client: GetClientBugItem | null;
    updatedBy: GetUserBugItem | null;
    priority: GetPriorityBugItem | null;
    status: GetStatusBugItem | null;

    assignments: GetAssignmentBugItem[];
    childJobs: GetJobLinkBugItem[];
    comments: GetCommentBugItem[];
    files: GetFileBugItem[];
    functionalities: GetFunctionalityBugItem[];
    parentJobs: GetJobLinkBugItem [];
    products: GetProductBugItem[];
  }
  
  export interface GetAssignmentBugItem {
    assignedDate: Date;
    assignedTo: string;
    assignedToFullName: string;
    assignedToEmail: string;
    assignedBy: string;
    assignedByFullName: string;
    assignedByEmail: string;
  }
  export interface GetClientBugItem {
    clientID: string;
    name: string;
    alias: string;
  }
  
  export interface GetCommentBugItem {
    commentID: number;
    comment: string;
    commentDate: Date;
    isEdited: boolean;
    commentedBy: string;
    commentedByEmail: string;
    commentedByFullName: string;
  }
  
  export interface GetFileBugItem {
    fileID: string; 
    name: string;
    size: number;
    extension: string;
    addedDate: Date;
    addedBy: string;
    addedByEmail: string;
    addedByFullName: string;
  }
  
  export interface GetFunctionalityBugItem {
    functionalityID: number;
    name: string;
    tag: string;
    tagColor: string | null;
    description: string;
  }
  
  export interface GetPriorityBugItem {
    priorityID: number,
    tag: string;
    tagColor: string | null;
    description: string | null;
  }

  export interface GetProductBugItem {
    productID: number;
    alias: string;
    name: string;
    description: string | null;
  }
  
  export interface GetStatusBugItem {
    statusID: number;
    tag: string;
    tagColor: string | null;
    description: string | null;
  }
  
  export interface GetTeamBugItem {
    alias: string;
    speciality: string;
    name: string;
    description: string;
  }
  
  export interface GetUserBugItem {
    userName: string;
    firstName: string;
    lastName: string;
    fullName: string;
    email: string;
  }

  export interface GetJobLinkBugItem {
    parentJobID: number;
    parentJobHeader: number;
    parentJobDescription: number;
    childJobID: number;
    childJobHeader: number;
    childJobDescription: number;
    linkDate: string;
    linkedBy: string;
    linkedByFullName: string;
    linkedByEmail: string;
  }


  